### PR TITLE
Node 701 group explorer queries

### LINKED ISSUE #701: Query for groups near a location sorted by closeness

### PR DESCRIPTION
**Pull request by thomasgwatson**
Closes #701

### CODE REVIEW

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -615,7 +615,8 @@ type Query {
     parentSlugs: [String],
     search: String,
     sortBy: String,
-    visibility: Int
+    visibility: Int,
+    coord: PointInput
```
Comment (tibetsprague) [inline api/graphql/schema.graphql:619]: let's alphabetize

Comment (tibetsprague) [reply api/graphql/schema.graphql:619]: also if this param always means "near this coordinate" then i wonder if we can come up with a more descriptive name. nearCoord? aroundCoord?

File: api/services/Search.js [hunk 2]
```diff
@@ -28,7 +38,7 @@ module.exports = {
       }
 
       if (opts.visibility) {
-        qb.where('groups.visibility', opts.visibility)
+        qb.whereIn('groups.visibility', opts.visibility)
```
Comment (tibetsprague) [inline api/services/Search.js:41]: does this work if opts.visibility is not an array? i want to make sure we dont break existing queries from mobile app

File: api/services/Search.js [hunk 3]
```diff
@@ -45,15 +55,16 @@ module.exports = {
...
+        boundingBox: opts.boundingBox
+      }, qb)
       // this counts total rows matching the criteria, disregarding limit,
       // which is useful for pagination
       countTotal(qb, 'groups', opts.totalColumnName)
-
       qb.limit(opts.limit)
       qb.offset(opts.offset)
-      qb.groupBy('groups.id')
+      if (!opts.coord && !opts.sort === 'size') {
```
Comment (tibetsprague) [inline api/services/Search.js:65]: a comment here would be good. even if you may not totally understand why

File: api/services/Search.js (hunk 2 above)
Comment (thomasgwatson) [reply api/services/Search.js:41]: There are three cases of this pattern, in forGroups, forTags and forPosts, and this was the only `where`. The other two are `whereIn`. @lorenjohnson do you recall if the mobile app has any group search queries, especially if they pass visibility? I figured we should standardize on array as input (although I think makeModels.js has a default value that I would need to change as well)

File: api/graphql/schema.graphql (hunk 1 above)
Comment (thomasgwatson) [reply api/graphql/schema.graphql]: Updated

**Review by tibetsprague — APPROVED** [review]

**Pull request merged**